src/lTerm_draw.ml

```diff
@@ -586,3 +586,10 @@ let draw_frame ctx rect ?style connection =
   draw_piece ctx (rect.row1 + 0) (rect.col2 - 1) ?style { top = Blank; bottom = connection; left = connection; right = Blank };
   draw_piece ctx (rect.row2 - 1) (rect.col2 - 1) ?style { top = connection; bottom = Blank; left = connection; right = Blank };
   draw_piece ctx (rect.row2 - 1) (rect.col1 + 0) ?style { top = connection; bottom = Blank; left = Blank; right = connection }
+
+let draw_frame_labelled ctx rect ?style ?(alignment=H_align_left) label connection = 
+  draw_frame ctx rect ?style connection;
+  try 
```

Comment:
I dunno.  `sub` does this error check already and uses an exception to indicate failure.

As a compromise I've added a `sub_opt` function which returns an option instead.  I suspect this should be the other way round (`sub` and `sub_exn`) but that would change the interface.